### DISCUSSION TITLE
bumping our mac requirements 10.13+

### DISCUSSION THREAD

**Discussion by thezoggy** [sabnzbd/sabnzbd#2784] [General]
Recently in chat we were talking about how everything these days pretty much expects macos 10.13+
> macOS 10.13 (high sierra) came out Sept 2017 with most recent patch 10.13.6 nov 2020.

However in sab we've done things to keep macos 10.9+ support.
> macOS 10.9 (mavericks) came out in 2013 and last patch was july 2016.

I'm not sure anyone running a mac with 8-11 years old os should be connecting to the internet.. 
Maybe if sab stopped supporting them it would be the final nudge for them to get around to updating (probably not, but they are going to struggle with so many other things already).

>> side note, while apple artificially limited newer software on some legacy hardware there is this patch to get around that for some people so they can get to 10.15: http://dosdude1.com/catalina/

Checking around;
- plex wants macOS 10.13+
- docker wants macOS 10.13+

Latest browsers actually want a little newer:
- firefox/chrome requires macOS 10.15+

Looking, I see we use macos-11 runner so we can use older xcode 13.x that does 10.9+ compatibility
https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md

I do wonder if it works on 10.9 though, I see during the builder requirements we do "--platform macosx_10_12_universal" which looks to make 10.12 version?

Thoughts on us switching to macos-12, which will have newer xcode 14.x which does 10.13+
https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

**Comment by Safihre** [comment]
I thought about it, but we don't really do anything special to support 10.9. Only compiling the pyinstaller bootloader ourselves. 
The 10.12 one is for cryptography, which is a non essential module.

**Comment by Safihre** [comment]
Python has dropped support for <10.13 in Python 3.12.6:
https://www.python.org/downloads/release/python-3126/

So we should now do the same.

**Reply by thezoggy** [reply]
i see the macos14-arm runner does have py3.12.6, the macos14 one still on 3.12.5 but pr exists -- https://github.com/actions/runner-images/pull/10615

**Reply by Safihre** [reply]
We download the Python.org installer in our build script.
